extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php

```diff
@@ -284,6 +284,34 @@ static public function onArticleViewHeader( Article $article, bool &$outputDone,
 		return true;
 	}
 
+	/**
+	 * When #REDIRECT is used make a proper 301 redirect
+	 *
+	 * @param Title $title
+	 * @param $unused
+	 * @param OutputPage $output
+	 * @param User $user
+	 * @param WebRequest $request
+	 * @param MediaWiki $mediawiki
+	 * @return bool
+	 */
+	static public function onBeforeInitialize(
+		Title $title, $unused, OutputPage $output,
+		User $user, WebRequest $request, MediaWiki $mediawiki
+	): bool {
+		if ( !$user->isAnon() || !$title->isRedirect() ) {
+			return true;
+		}
+
+		$queryParams = $request->getQueryValues();
+		unset( $queryParams['title'] );
+		$targetUrl = $output->getWikiPage()->getRedirectTarget()->getFullURL( $queryParams );
+
+		$output->redirect( $targetUrl, HTTP_REDIRECT_PERM );
```

Comment:
Hmm, actually, is `HTTP_REDIRECT_PERM` definitely defined in our stack? IIRC it's part of a pecl extension that we don't have and a quick test suggests it's not defined:
```
$ run_maintenance --id=147 --script="eval.php"
> var_dump(defined('HTTP_REDIRECT_PERM'));
bool(false)

> echo HTTP_REDIRECT_PERM;
HTTP_REDIRECT_PERM
```

Using constants in place of all these redirect codes we use is probably worthwhile though.

Reply:
But it works... :speak_no_evil: Yeah, I'll update it, might break at some point.